src/analysis/complex/cauchy_integral.lean

```diff
@@ -0,0 +1,482 @@
+      rw circle_integral.integral_sub; refine (hc'.smul _).circle_integrable hr0.le,
+      { exact hc.mono (subset_inter (sphere_subset_closed_ball.trans $
+          closed_ball_subset_closed_ball hrR) hzne) },
+      { exact continuous_on_const }
+    end
+  ... ≤ 2 * π * r * (r⁻¹ * (ε / (2 * π))) :
+    begin
+      refine circle_integral.norm_integral_le_of_norm_le_const hr0.le (λ z hz, _),
+      specialize hzne z hz,
+      rw [mem_sphere, dist_eq_norm] at hz,
+      rw [norm_smul, normed_field.norm_inv, hz, ← dist_eq_norm],
+      refine mul_le_mul_of_nonneg_left (hδ _ ⟨_, hzne⟩).le (inv_nonneg.2 hr0.le),
+      rwa [mem_closed_ball_iff_norm, hz]
+    end
+  ... = ε : by { field_simp [hr0.ne', real.two_pi_pos.ne'], ac_refl }
+end
+
+/-- **Cauchy integral formula** for the value at the center of a disc. If `f` is differentiable on a
+closed disc of radius `R`, then the integral
+$\int_{|z|=R} f(z)\,d(\arg z)=-i\int_{|z|=R}\frac{f(z)\,dz}{z}$ is equal to $2πy`. Moreover, the
+same is true if at the points of some countable set, `f` is only continuous, not differentiable. -/
+lemma circle_integral_sub_center_inv_smul_of_differentiable_on_off_countable {R : ℝ} (h0 : 0 < R)
+  {f : ℂ → E} {c : ℂ} {s : set ℂ} (hs : countable s)
+  (hc : continuous_on f (closed_ball c R)) (hd : ∀ z ∈ ball c R \ s, differentiable_at ℂ f z) :
+  ∮ z in C(c, R), (z - c)⁻¹ • f z = (2 * π * I : ℂ) • f c :=
+circle_integral_sub_center_inv_smul_of_differentiable_on_off_countable_of_tendsto h0 hs
+  (hc.mono $ diff_subset _ _) (λ z hz, hd z ⟨hz.1.1, hz.2⟩)
+  (hc.continuous_at $ closed_ball_mem_nhds _ h0).continuous_within_at
+
+/-- **Cauchy theorem**: the integral of a complex differentiable function over the boundary of a
+disc equals zero. Moreover, the same is true if at the points of some countable set, `f` is only
+continuous. -/
+lemma circle_integral_eq_zero_of_differentiable_on_off_countable {R : ℝ} (h0 : 0 ≤ R) {f : ℂ → E}
+  {c : ℂ} {s : set ℂ} (hs : countable s) (hc : continuous_on f (closed_ball c R))
+  (hd : ∀ z ∈ ball c R \ s, differentiable_at ℂ f z) :
+  ∮ z in C(c, R), f z = 0 :=
+begin
+  rcases h0.eq_or_lt with rfl|h0, { apply circle_integral.integral_radius_zero },
+  calc ∮ z in C(c, R), f z = ∮ z in C(c, R), (z - c)⁻¹ • (z - c) • f z :
+    begin
+      refine circle_integral.integral_congr h0.le (λ z hz, (inv_smul_smul₀ (λ h₀, _) _).symm),
+      rw [mem_sphere, dist_eq, h₀, abs_zero] at hz,
+      exact h0.ne hz
+    end
+  ... = (2 * ↑π * I : ℂ) • (c - c) • f c :
+    circle_integral_sub_center_inv_smul_of_differentiable_on_off_countable h0 hs
+      ((continuous_on_id.sub continuous_on_const).smul hc)
+      (λ z hz, (differentiable_at_id.sub_const _).smul (hd z hz))
+  ... = 0 : by rw [sub_self, zero_smul, smul_zero]
+end
+
+/-- **Cauchy integral formula**: if `f : ℂ → E` is complex differentiable on a closed disc of radius
+`R`, then for any `w` in the corresponding open disc we have
+$\oint_{|z-c|=R}(z-w)^{-1}f(z)\,dz=2\pi i\,f(w)$.
+-/
```

Comment:
Could you add the docstring to the non-aux version (allowing `w` in `s`) and mention here the difference between the aux and the non-aux versions?